File: learning/cs50/pset6/cash/cash.py
# Asking an owe to the user:

while True:
    n = float(input("Change owed: "))
    if n >= 0:
        break

# Coins available

coins = [25, 10, 5, 1]

# Counter and total coins
owe = round(n*100)
counter = 0

# Checking coins quantity

while owe >= coins[0]:
    owe -=coins[0]
    counter += 1

while owe >= coins[1]:
    owe -=coins[1]
    counter += 1

while owe >= coins[2]:
    owe -=coins[2]
    counter += 1

while owe >= coins[3]:
    owe -=coins[3]
    counter += 1

print(counter)

File: learning/cs50/pset1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main (void)
{
    // Asking an owe value to the user
    float n;
    do
    {
        n = get_float("Change owed: ");
    }
    while (n <= 0);

    // Rounding value
    int owe = round (n*100);

    // Coins available
    int a = 25;
    int b = 10;
    int c = 5;
    int d = 1;

            // Running program

    // Counter to number of coins
    int coins = 0;

            // Checking the number of coins to owe

    // Check 0.25 coins

    while (owe >= a)
    {
        owe -= a;
        coins++;
    }

    // Check 0.10 coins

    while (owe >= b)
    {
        owe -= b;
        coins++;
    }

    // Check 0.05 coins

    while (owe >= c)
    {
        owe -= c;
        coins++;
    }

    // Check 0.01 coins

    while (owe >= d)
    {
        owe -= d;
        coins++;
    }

    // Printing coins on the screen
    printf("%i\n", coins);

}